### PR TITLE
Prepare concept for the Application Connector improvements

### PR DESCRIPTION
**Pull request by lszymik**
**Description**

Changes proposed in this pull request:

- A concept for OData support in the Application Connector
- A concept for the Application Connector modularization

**Related issue(s)**

### CODE REVIEW

**Comment by akrausesap** [comment]
You might want to specify an OData Version as this has an effect on the Service. Also you might want to omit the $metadata as this is also covered by the specification.

**Comment by brendandobbs** [comment]
Looks good. 

In some cases we may need to replace FQ'ed URLs for the target environment to a URL to the gateway/proxy for the document that gets stored in Minio (we also need to do this for Swagger/Open API).

**Comment by brendandobbs** [comment]
For v2 services the metadata document will be in XML format, for v4 services JSON is the default (both support JSON for the payload). The schema is also quite different between v2 and v4. If the purpose of storing the document is only for displaying the service in the Service Catalog it might make sense to transform the document into a standard/common format before saving it to Minio

**Comment by lszymik** [comment]
@brendandobbs Thank you very much for your input. Regarding 

> In some cases we may need to replace FQ'ed URLs for the target environment to a URL to the gateway/proxy for the document that gets stored in Minio

It is a very good point, I will add it to the concept. We are already doing it for OpenAPI specifications.

The stored specification will be used in the future for a client generation. I am not sure if we can normalize it now, as it might come as a surprise for a developer. I will check with Service Catalog UI experts.

**Comment by akrausesap** [comment]
There are more things that came to my mind: 

- OData services are often very small in business scope, hence it might be good to be able to register multiple OData Services in one call to avoid a bloated service catalog

- CSRF Token Handling should be indicated. Especially as the connector manages session and authentication it is probabaly impossible to handle for  the developer of the lambda

Cheers

Andreas

**Comment by akrausesap** [comment]
@lszymik @brendandobbs: Ralf Handl (whom we are meeting next week) has created an XSLT to transform v2 metadata to v4 which we might be able to use if normalization is needed

File: sig-and-wg/sig-core/proposals/application-connector-as-addon.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+- Nginx-Ingress-Controller is installed as a sub-chart of the core Helm chart
+- Application Connector is installed as a sub-chart of the core Helm chart (Metadata Service and Connector Service)
+- Remote Environment CRD is installed separately as cluster-essential
+- Remote Environments, like ec-default, are installed as a separate Helm chart
+
+## New approach
+
+### Helm chart with the Application Connector
+
+There will be a separate Helm chart which will install all necesarry items:
```
Comment (tomekpapiernik) [inline sig-and-wg/sig-core/proposals/application-connector-as-addon.md]: necesarry -> necessary

File: sig-and-wg/sig-core/proposals/application-connector-as-addon.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+- Remote Environment CRD
+- Nging-Ingress controller
```
Comment (tomekpapiernik) [inline sig-and-wg/sig-core/proposals/application-connector-as-addon.md]: Nging  -> Nginx

File: sig-and-wg/sig-core/proposals/application-connector-as-addon.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- Connector service
+- Metadata service
+- Remote Environment broker
+- A Remote Environment operator
+- UI for Application Connector need to be added as well. It must be discussed with Swinka team.
+
+The helm chart can be used manually and can be integrated with Kyma addons, Kyma bundles for service catalog or kyma core as gardener addon, [Bouquet](https://github.com/gardener/bouquet).
```
Comment (tomekpapiernik) [inline sig-and-wg/sig-core/proposals/application-connector-as-addon.md]: capitalize "Kyma" and "Gardener"

Comment (tomekpapiernik) [inline sig-and-wg/sig-core/proposals/application-connector-as-addon.md]: integrated with Kyma addons -> integrated with Kyma add-ons

File: sig-and-wg/sig-core/proposals/application-connector-as-addon.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+### Remote Environment Operator
+
+There will be a new Remote Environment operator which will replace Helm charts for RE. The operator will react to modification of the RE Custom Resource and will add all required items, like Event service, Proxy service.
+
+### Use caseses
```
Comment (tomekpapiernik) [inline sig-and-wg/sig-core/proposals/application-connector-as-addon.md]: Use cases

File: sig-and-wg/sig-core/proposals/odata-support.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+### API Registration
+
+The Metadata service will be extended with the possibility to register OData API. Therefore, the following changes will be introduced:
+
+- Current registration of API will be the same with inlined API specification (an in-line version has higher priority).
+- Registration of API with `specUrl` will be added. If `spec` fields is not given, the API definition will be downloaded from `specUrl`.
+- API registration will have an additional optional field type, where a user will be able to mark API as OData one or classic REST API. Type can be stored in tags field which is already part of the RE CRD.
+- if the type is provided then the API specification will be fetched from `specUrl` or an in-line version will be used for REST and fetching the $metadata will be used for OData API
+- If the type is not provided and there is no `spec` or `specUrl` given, the Metadata service will do a lookup to the API URL/$metadata and automatically set the API type to OData one. OData registration will not have specification fields.
+- OData specification will be stored in Minio and need to have the FQ'ed URLs ammended to the Gateway specific ones.
```
Comment (tomekpapiernik) [inline sig-and-wg/sig-core/proposals/odata-support.md]: FQ'ed URLs ammended -> amended

File: sig-and-wg/sig-core/proposals/odata-support.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+- Implementation must not introduce any breaking changes for already implemented integrations.
+
+#### API snippet
+
+Example API calls with a OData API or a Rest API.
```
Comment (tomekpapiernik) [inline sig-and-wg/sig-core/proposals/odata-support.md]: with a OData API ->  with an OData API

**Pull request merged**